pkg/management/component.go

```diff
@@ -488,25 +488,55 @@ func (c *component) AddClientRolesToUser(ctx context.Context, realmName, userID,
 	return err
 }
 
-func (c *component) ResetPassword(ctx context.Context, realmName string, userID string, password api.PasswordRepresentation) error {
+func (c *component) ResetPassword(ctx context.Context, realmName string, userID string, password api.PasswordRepresentation) (string, error) {
 	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
 
+	var pwd string
+	var err error
 	var credKc kc.CredentialRepresentation
 	var passwordType = "password"
 	credKc.Type = &passwordType
-	credKc.Value = password.Value
 
-	err := c.keycloakClient.ResetPassword(accessToken, realmName, userID, credKc)
+	if password.Value == nil {
+		// no password value was provided; a new password, that respects the password policy of the realm, will be generated
+		var minLength int = 12
+
+		//obtain password policy
+		realmKc, err := c.keycloakClient.GetRealm(accessToken, realmName)
+		if err == nil {
+			if realmKc.PasswordPolicy == nil {
```

Comment:
I would create an internal.GeneratePasswordXXX(policy string*, minLength int) that would try to generate passwords (while pwd==userID AND err==nil) and decide to call GeneratePassword or GeneratePasswordFromKeycloakPolicy according to the given policy. It would be shorter here in component.go

Reply:
DONE

Reply:
DONE